### PR TITLE
Get Java11 compatible #127

### PR DESCRIPTION
**Pull request by EugenMayer**
- upgrade gradle to 5.2.2 and remove gradle install wrapper task, it is
- all tests are good (macosx and linux)
- also officeconverter does run with jodconverter on java11

### CODE REVIEW

**Comment by sbraconnier** [comment]
Isn't it an upgrade to gradle 5.2.1 instead ?

**Comment by EugenMayer** [comment]
Yes sorry, mixed it up its 5.1.2
I also upgraded spring-boot to 2.1.2.RELEASE due to some adjustments there for j11 - right now failing to build due to javadoc

```
javadoc: error - The code being documented uses modules but the packages defined in https://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.
```

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jodconverter-samples:jodconverter-sample-spring-boot:javadoc'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): '/tmp/jodconverter/jodconverter-samples/jodconverter-sample-spring-boot/build/tmp/javadoc/javadoc.options'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

My tests are yet not representative, i had them run on 1.8 - testing 11.0 now

**Comment by EugenMayer** [comment]
The javadoc issue is probably related to https://bugs.openjdk.java.net/browse/JDK-8212233

@sepe81 we could just squash the whole PR later one and reduce it just to "j11 upgrade", or do you really want me to rebase that one commit?

**Comment by EugenMayer** [comment]
Ok tests are ok on 1.8 with gradle 5.1.2 but failing with java11

all 27 cli tests are failing
```
org.jodconverter.cli.ConvertTest > classMethod FAILED
    org.mockito.exceptions.base.MockitoException at ConvertTest.java:58
        Caused by: java.lang.UnsupportedOperationException at ConvertTest.java:58
            Caused by: java.lang.IllegalStateException
                Caused by: java.lang.NoSuchMethodException
```

Seems like this is probably a mockito issue, which is stuck at 2.9.0 because of powermockitio

```
// Latest version -> https://mvnrepository.com/artifact/org.mockito/mockito-core
// 2.9.0 is the latest version compatible powermock version 1.7.3 so we must
// override the version provided by spring boot (see gradle/java-projects.gradle).
// See https://stackoverflow.com/questions/47240116/mockito-and-powermockito-error
 mockito_version : '2.9.0',
```
Not worked at all at powermockito, any comments?

**Comment by sepe81** [comment]
@EugenMayer Squashing later seems fine. :+1: 

Additionally I found a related open issue/suggestion for gradle: https://github.com/gradle/gradle/issues/8267

**Comment by EugenMayer** [comment]
move travis to openjdk11 but i probably would suggest doing a matrix build for the lowest version we would aim to, so jdk8 too? Thoughts?

**Comment by sepe81** [comment]
@EugenMayer Maybe https://stackoverflow.com/a/34874587 helps as a workaround for the javadoc problem?

**Comment by EugenMayer** [comment]
@sepe81 @sbraconnier is 

```
subprojects {
    tasks.withType(Javadoc).all { enabled = false }
}
```

really what we want, i mean in terms do we want to disable those?

**Comment by EugenMayer** [comment]
Ok tests are failing properly now and reflect my javadoc issue, thats good :)

**Comment by sepe81** [comment]
> really what we want, i mean in terms do we want to disable those?

@EugenMayer As I wrote -- it's just a workaround, maybe till there is a real solution for gradle with Java 11

**Comment by EugenMayer** [comment]
@sepe81 https://stackoverflow.com/a/34874587 would work, but i guess that is not want is wanted - i need some more details of @sbraconnier on this.

Still the cli tests are failing, are you both interested in working on this in general?

**Comment by sbraconnier** [comment]
Wow thanks for all the good work! I won't be able to check this until next week but I definitely want to get Java 11 compatibility! I'll get back to you.

File: .travis.yml [hunk 1]
```diff
@@ -7,17 +7,34 @@ dist: trusty
 matrix:
   include:
     - os: linux
-      jdk: oraclejdk8
+      jdk: oraclejd8
```
Comment (CauanCabral) [inline .travis.yml]: Is that a typo?

Comment (sepe81) [reply .travis.yml]: most likely, yes. good catch :)

Comment (EugenMayer) [reply .travis.yml]: Wasn't me! :)

**Comment by sbraconnier** [comment]
I finally had time to work on this! To summarize, I updated the PowerMock library, I changed the javadoc options in the `gradle/java-projects.gradle` file (to check the current java version) and I added the `-x javadoc` switch in the `.travis.yml` file to skip the javadoc task when building using java11.

**Comment by EugenMayer** [comment]
Thank you for your hard work once again!

**Comment by EugenMayer** [comment]
@sbraconnier just wanted to confirm that my j11 based builds are working now ( at https://github.com/EugenMayer/docker-image-jodconverter ) so it actually worked out well :)

**Comment by sbraconnier** [comment]
Glad to hear that!!  👍

**Pull request merged**